src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -252,3 +270,28 @@ def _init_seen_tables(self):
 
     def _table_already_upgraded(self, target) -> bool:
         return target in self._seen_tables
+
+    def revert_migrated_tables(self, schema: str | None = None, table: str | None = None):
+        upgraded_tables = []
+        for cur_table in self._tc.snapshot():
+            schema_match = not schema or cur_table.database == schema
+            table_match = not table or cur_table.name == table
+            if schema_match and table_match and cur_table.upgraded_to is not None:
```

Comment:
I suspect there's a hidden bug here, that would be difficult to catch. Please rewrite this as two separate if-continue statements, otherwise it's not supportable and may result in customer data loss

Reply:
Fixed